app/src/main/java/com/uc/mymovies/ListAdapter.kt

```diff
@@ -12,42 +12,37 @@ import com.uc.mymovies.dto.Movie
 
 class ListAdapter(private val context: Context, var movies: ArrayList<Movie>):BaseAdapter() {
 
-    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup?) : View {
         var view: View? = convertView
         val viewHolder: ViewHolder
-
-        if (view == null) {
-            viewHolder = ViewHolder()
+        view?.let {
             val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
             view = inflater.inflate(R.layout.list_item, null, true)
-
-            viewHolder.name = view.findViewById(R.id.name)
-            viewHolder.image = view.findViewById(R.id.image)
-            viewHolder.favoriteImage = view.findViewById(R.id.favorite)
-
+            viewHolder = ViewHolder().apply {
+                name = view.findViewById(R.id.name)
+                image = view.findViewById(R.id.image)
+                favoriteImage = view.findViewById(R.id.favorite)
+            }
             view!!.tag = viewHolder
```

Comment:
use apply to group object
https://phauer.com/2017/idiomatic-kotlin-best-practices/